.github/workflows/ci.yaml

```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  rust:
+    name: Rust format, lint, and test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      # https://github.com/actions-rs/toolchain
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
```

Comment:
the rust version is specified in the file in the repo root.. I assume that should be used here?

Reply:
No because this is just the version of rustup toolchain (stable, nightly or a specific version).

Reply:
By default it will read the version from the toolchain file but we just need to rename it (pushing now).

https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file

Reply:
ahh it installs it on fresh ubuntu... would be interesting to see if it has any chaching...